controller/src/main/java/io/pravega/controller/server/SegmentHelper.java

```diff
@@ -635,17 +651,34 @@ private void closeConnection(Reply reply, RawClient client) {
     }
 
     private <T extends Request & WireCommand> CompletableFuture<Reply> sendRequest(RawClient connection, long requestId, T request) {
-        return connection.sendRequest(requestId, request)
+        CompletableFuture<Reply> future = Futures.futureWithTimeout(timeout.get(), executorService);
+        AtomicReference<CompletableFuture<Reply>> f = new AtomicReference<>();
+        Futures.completeAfter(() -> {
```

Comment:
Can we create a method on lines of the below pattern? This would eliminate the need for extra code.
 ```
 public static <T> CompletableFuture<T> futureWithTimeout(CompletableFuture<T> cf, Duration timeout, ScheduledExecutorService executorService) {
      
        ScheduledFuture<Boolean> sf = executorService.schedule(() -> cf.completeExceptionally(new TimeoutException(tag)), timeout.toMillis(), TimeUnit.MILLISECONDS);
        result.whenComplete((r, ex) -> sf.cancel(true));
        return cf;
    }
```

Reply:
Interesting JAVA9 APIS support this https://docs.oracle.com/javase/9/docs/api/java/util/concurrent/CompletableFuture.html#completeOnTimeout-T-long-java.util.concurrent.TimeUnit-